server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/MinorCompactor.java

```diff
@@ -93,7 +94,18 @@ public CompactionStats call() {
     try {
       do {
         try {
-          CompactionStats ret = super.call();
+          CompactionStats ret = null;
+          try {
+            ret = super.call();
+          } catch (Exception e) {
+            if (tabletServer.getLock() == null || !tabletServer.getLock().verifyLockAtSource()) {
```

Comment:
Is there a small race condition here with calling getLock() twice? I'm not sure if it's possible for the state to really change in this case between reads but I was wondering if it's better to do something like 

```java
var lock = tabletServer.getLock();
if (lock == null || !lock.verifyLockAtSource()) {
```

Reply:
Updated in 1d9e34f